safesocket/safesocket_darwin.go

```diff
@@ -17,23 +20,246 @@ import (
 	"sync"
 	"time"
 
+	"golang.org/x/sys/unix"
 	"tailscale.com/version"
 )
 
 func init() {
 	localTCPPortAndToken = localTCPPortAndTokenDarwin
 }
 
-// localTCPPortAndTokenMacsys returns the localhost TCP port number and auth token
+// SameUserProofTokenLength is the length in bytes of the token used to
+// authenticate safesocket connections
+const (
+	sameUserProofTokenLength = 10
+	SyxExtLibraryDir         = "/Library/Tailscale"
+)
+
+type darwinCredentials struct {
+	mu    sync.Mutex
+	creds struct {
+		token string
+		port  int
+	}
+}
+
+var (
+	// explicit credentials set by SetCredentials
+	dc darwinCredentials
+	// file descriptor to the open sameuserproof file for mac App Store
+	sameuserproofFD *os.File
+
+	// For testing only to force macsys
+	isMacSysExt = version.IsMacSysExt()
```

Comment:
make it a func var, not a bool var (drop the parens), so the work can still be lazy and doesn't add init-time load to all users who link this package